library/test/src/formatters/junit.rs

```diff
@@ -26,6 +26,18 @@ impl<T: Write> JunitFormatter<T> {
     }
 }
 
+fn str_to_cdata(s: &str) -> String {
+    // Drop the stdout in a cdata. Unfortunately, you can't put either of `]]>` or
+    // `<?'` in a CDATA block, so the escaping gets a little weird.
+    let escaped_output = s.replace("]]>", "]]]]><![CDATA[>");
+    let escaped_output = escaped_output.replace("<?", "<]]><![CDATA[?");
+    // We also smuggle newlines as &#xa so as to keep all the output on one line
+    let escaped_output = escaped_output.replace("\n", "]]>&#xA;<![CDATA[");
```

Comment:
The compiler already pulls in aho-corasick, but the test library does not. Failing test stdout could easily balloon to gigantic sizes -- should I write a PR to swap these replaces to use aho-corasick?

Reply:
I don't think so -- it's a higher bar to add dependencies to the standard library, including `test`.

Reply:
discussion on Zulip: Let's wait until someone does a perf measurement and finds this to blame.